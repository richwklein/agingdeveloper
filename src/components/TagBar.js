import React from "react";
import {Link} from "gatsby";
import {Button, Grid} from "@material-ui/core";
import {Folder, LocalOffer} from "@material-ui/icons";
import kebabCase from "lodash/kebabCase";

const TagsBox = ({category, tags}) => {
  return (
    <Grid container spacing={1}>
      <Grid item key={category}>
        <Button
          startIcon={<Folder />}
          variant="text"
          edge="start"
          key={category}
          component={Link}
          to={`/category/${kebabCase(category)}`} >
          {category}
        </Button>
      </Grid>
      {tags.map((tag) => {
        return (
          <Grid item key={tag}>
            <Button
              startIcon={<LocalOffer />}
              variant="text"
              edge="start"
              key={tag}
              component={Link}
              to={`/tag/${kebabCase(tag)}`} >
              {tag}
            </Button>
          </Grid>
        );
      })}
    </Grid>
  );
};

export default TagsBox;
