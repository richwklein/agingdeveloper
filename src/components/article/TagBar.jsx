import React from "react";
import {Button, Grid} from "@mui/material";
import InternalLink from "../common/InternalLink";
import TagOutlinedIcon from "@mui/icons-material/TagOutlined";
import PropTypes from "prop-types";
import slug from "slug";

export const Tag = ({name}) => {
  return (
    <Button
      startIcon={ <TagOutlinedIcon />}
      variant="text"
      component={InternalLink}
      to={`/tag/${slug(name)}`}
      sx={{
        "& .MuiButton-startIcon": {
          marginRight: .35,
        },
      }}>
      {name}
    </Button>
  );
};

Tag.propTypes = {
  name: PropTypes.string.isRequired,
};

const TagBar = ({tags}) => {
  return (
    <Grid container spacing={.5}>
      {tags.map((tag) => {
        return (
          <Grid item key={tag}>
            <Tag name={tag} />
          </Grid>
        );
      })}
    </Grid>
  );
};

TagBar.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.string),
};

export default TagBar;
