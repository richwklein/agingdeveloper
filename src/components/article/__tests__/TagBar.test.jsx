import React from "react";
import {cleanup, render} from "@testing-library/react";
import TagBar, {Tag} from "../TagBar";

// TODO improve tests
describe("TagBar", () => {
  afterEach(cleanup);

  const renderComponent = () => render(
      <TagBar tags={["Theme Park", "Genie+", "Sadness", "I â™¥ Dogs"]} />,
  );

  test("that it renders correctly.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("that a tag is generated correctly.", () => {
    const {container} = render(<Tag name="Tag Name" />);
    expect(container.firstChild).toContainHTML("data-testid=\"TagOutlinedIcon\"");
    expect(container.firstChild).toContainHTML("href=\"/tag/tag-name\"");
  });
});
