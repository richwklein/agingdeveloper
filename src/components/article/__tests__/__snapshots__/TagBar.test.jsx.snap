// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagBar that it renders correctly. 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-0.5 css-125mmfg-MuiGrid-root"
>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-a6ypk3-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
      data-link="internal"
      href="/tag/theme-park"
      tabindex="0"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="TagOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 10V8h-4V4h-2v4h-4V4H8v4H4v2h4v4H4v2h4v4h2v-4h4v4h2v-4h4v-2h-4v-4h4zm-6 4h-4v-4h4v4z"
          />
        </svg>
      </span>
      Theme Park
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-a6ypk3-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
      data-link="internal"
      href="/tag/genie"
      tabindex="0"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="TagOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 10V8h-4V4h-2v4h-4V4H8v4H4v2h4v4H4v2h4v4h2v-4h4v4h2v-4h4v-2h-4v-4h4zm-6 4h-4v-4h4v4z"
          />
        </svg>
      </span>
      Genie+
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-a6ypk3-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
      data-link="internal"
      href="/tag/sadness"
      tabindex="0"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="TagOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 10V8h-4V4h-2v4h-4V4H8v4H4v2h4v4H4v2h4v4h2v-4h4v4h2v-4h4v-2h-4v-4h4zm-6 4h-4v-4h4v4z"
          />
        </svg>
      </span>
      Sadness
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-a6ypk3-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
      data-link="internal"
      href="/tag/i-dogs"
      tabindex="0"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="TagOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 10V8h-4V4h-2v4h-4V4H8v4H4v2h4v4H4v2h4v4h2v-4h4v4h2v-4h4v-2h-4v-4h4zm-6 4h-4v-4h4v4z"
          />
        </svg>
      </span>
      I â™¥ Dogs
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
